src/Dapr.Workflow/WorkflowState.cs

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.Workflow
+{
+    using System;
+    using Microsoft.DurableTask.Client;
+
+    /// <summary>
+    /// Represents a snapshot of a workflow instance's current state, including runtime status.
+    /// </summary>
+    public class WorkflowState
+    {
+        readonly OrchestrationMetadata? workflowState;
+
+        internal WorkflowState(OrchestrationMetadata? workflowState)
+        {
+            // This value will be null if the workflow doesn't exist.
+            this.workflowState = workflowState;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow instance exists.
+        /// </summary>
+        public bool Exists => this.workflowState != null;
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow is in a running state.
+        /// </summary>
+        public bool IsWorkflowRunning => this.workflowState?.RuntimeStatus == OrchestrationRuntimeStatus.Running;
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow is in a terminal state.
+        /// </summary>
+        public bool IsWorkflowCompleted => this.workflowState?.IsCompleted == true;
+
+        /// <summary>
+        /// Gets the time at which this workflow instance was created.
+        /// </summary>
+        public DateTimeOffset CreatedAt => this.workflowState?.CreatedAt ?? default;
+
+        /// <summary>
+        /// Gets the time at which this workflow instance last had its state updated.
+        /// </summary>
+        public DateTimeOffset LastUpdatedAt => this.workflowState?.LastUpdatedAt ?? default;
+
+        /// <summary>
+        /// Gets the execution status of the workflow.
+        /// </summary>
+        public WorkflowRuntimeStatus RuntimeStatus
+        {
+            get
+            {
+                if (this.workflowState == null)
+                {
+                    return WorkflowRuntimeStatus.Unknown;
+                }
+
+                switch (this.workflowState.RuntimeStatus)
+                {
+                    case OrchestrationRuntimeStatus.Running:
+                        return WorkflowRuntimeStatus.Running;
+                    case OrchestrationRuntimeStatus.Completed:
+                        return WorkflowRuntimeStatus.Completed;
+                    case OrchestrationRuntimeStatus.Failed:
+                        return WorkflowRuntimeStatus.Failed;
+                    case OrchestrationRuntimeStatus.Terminated:
+                        return WorkflowRuntimeStatus.Terminated;
+                    case OrchestrationRuntimeStatus.Pending:
+                        return WorkflowRuntimeStatus.Pending;
+                    default:
+                        return WorkflowRuntimeStatus.Unknown;
+                }
```

Comment:
So the only thing this is doing is:

* null => Unknown
* Suspended => Unknown
* everything else is returned as is. Is that it?

I am trying to understand if this couldn't be replaced with `return this.workflowState?.RuntimeStatus ?? WorkflowRuntimeStatus.Unknown` and the `Suspended` state is the only thing that would break this, so I am trying to understand `Suspended` was omitted from that `switch` on purpose or by accident.

Reply:
Ok, just noticed that this is doing a type mapping from `OrchestrationRuntimeStatus` to `WorkflowRuntimeStatus`. Anyway, re-checking if `Suspended` => should be equivalent to `Unknown`.

Reply:
Yes, so I unfortunately the `1.0.0-rc.1` version of the Durable Task SDK doesn't have the `Suspended` enum yet, so I can't make that change until the `1.0.0` version gets published. So for now, `WorkflowRuntimeStatus.Suspended` is unused until probably the next release.